broadcaster/utxo_preparer.go

```diff
@@ -0,0 +1,200 @@
+package broadcaster
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"math"
+	"time"
+
+	"github.com/bitcoin-sv/arc/lib/keyset"
+	"github.com/bitcoin-sv/arc/metamorph/metamorph_api"
+	"github.com/libsv/go-bt/v2"
+	"github.com/libsv/go-bt/v2/bscript"
+	"github.com/libsv/go-bt/v2/unlocker"
+)
+
+const (
+	maxOutputsDefault = 100
+	batchSizeDefault  = 20
+)
+
+type UtxoClient interface {
+	GetUTXOs(mainnet bool, lockingScript *bscript.Script, address string) ([]*bt.UTXO, error)
+}
+
+type UTXOPreparer struct {
+	logger            *slog.Logger
+	Client            ArcClient
+	FromKeySet        *keyset.KeySet
+	ToKeySet          *keyset.KeySet
+	Outputs           int64
+	SatoshisPerOutput uint64
+	IsTestnet         bool
+	CallbackURL       string
+	FeeQuote          *bt.FeeQuote
+	UtxoClient        UtxoClient
+
+	maxOutputs int
+	batchSize  int
+}
+
+func WithFees(miningFeeSatPerKb int) func(preparer *UTXOPreparer) {
+	return func(preparer *UTXOPreparer) {
+		var fq = bt.NewFeeQuote()
+
+		newStdFee := *stdFeeDefault
+		newDataFee := *dataFeeDefault
+
+		newStdFee.RelayFee.Satoshis = miningFeeSatPerKb
+		newDataFee.RelayFee.Satoshis = miningFeeSatPerKb
+
+		fq.AddQuote(bt.FeeTypeData, &newStdFee)
+		fq.AddQuote(bt.FeeTypeStandard, &newDataFee)
+
+		preparer.FeeQuote = fq
+	}
+}
+
+func WithBatchSize(batchSize int) func(preparer *UTXOPreparer) {
+	return func(preparer *UTXOPreparer) {
+		preparer.batchSize = batchSize
+	}
+}
+
+func WithMaxOutputs(maxOutputs int) func(preparer *UTXOPreparer) {
+	return func(preparer *UTXOPreparer) {
+		preparer.maxOutputs = maxOutputs
+	}
+}
+
+func NewUTXOPreparer(logger *slog.Logger, client ArcClient, fromKeySet *keyset.KeySet, toKeyset *keyset.KeySet, utxoClient UtxoClient, opts ...func(p *UTXOPreparer)) *UTXOPreparer {
+	utxoPreparer := &UTXOPreparer{
+		logger:     logger,
+		Client:     client,
+		FromKeySet: fromKeySet,
+		ToKeySet:   toKeyset,
+		IsTestnet:  true,
```

Comment:
Should that be hardcoded? or even const?

Reply:
That is just the default value. But I'll put it in a const